conans/client/build/autotools_environment.py

```diff
@@ -152,6 +152,11 @@ def configure(self, configure_dir=None, args=None, build=None, host=None, target
 
         triplet_args = []
 
+        if hasattr(self._conanfile, "conf"):
+            build = self._conanfile.conf["tools.autotoolsbuildenvironment"].build or build
```

Comment:
I am not sure we want to use the new conf to solve this bug. I would prefer not to touch the existing build helpers, but any new effort should be focused on moving things to the new ``conan.tools`` space, in this case ``conan.tools.gnu`` I guess.

Reply:
there is only `MakeToolchain` in tools.gnu. we probably need new build helper first then.

Reply:
Yes, exactly, ``conan.tools.gnu.AutotoolsToolchain`` could be the first piece:

Same criteria as the others:
- Not using os_build, arch_build
- Simplifying things and getting rid of legacy
- generate() as the way to go

In this case it should probably use something like virtualenvs.

Efforts should go into building this. I am not saying that a bad bug cannot be fixed on the previous ones, but I would like not to pollute the new "clean" conf mechanism for the old helpers.

Reply:
Lets do first https://github.com/conan-io/conan/pull/8457